src/polycubed/src/server/Resources/Endpoint/Resource.cpp

```diff
@@ -33,4 +46,152 @@ Operation Resource::OperationType(bool update, bool initialization) {
+              }
+              (*el).push_back(body);
+            } else { /* Otherwise it is element of an element of that array */
+              for (int j = 0; j < el->size(); j++) { /* find the interested element */
+                if ((*el)[j]["name"] == resItem[i + 1]) {
+                  el = &(*item)[resItem[i]][j];
+                  i++;
+                  break;
+                }
+              }
+            }
+          } else { /* Otherwise it is a creation/replacement of an element */
+            if (i == resItem.size() - 2) {
+              (*item)[resItem[i]] = body;
+            }
+          }
+          item = el;
+        }
+      }
+      break;
+    }
+
+    case Operation::kUpdate: {
+      nlohmann::json *item = &cubesConfig[cubeName];
+      for (int i = 2; i < resItem.size() - 1; i++) {
+        auto *el = &(*item)[resItem[i]];
+        if (el->is_array()) { /* If it is an element of an array, then it is an element of an array */
+          for (int j = 0; j < el->size(); j++) { /* Find the interested element of the array */
+            if ((*el)[j]["name"] == resItem[i + 1]) {
+              el = &(*item)[resItem[i]][j];
+              i++;
+              break;
+            }
+          }
+        }
+        item = el;
+      }
+
+      (*item)[resItem[resItem.size() - 1]] = body;
+      break;
+    }
+
+    case Operation::kDelete: {
+      if (resItem.size() == 2) { /* If it regards the full service */
+        cubesConfig.erase(cubeName);
+      } else { /* If it regards an element of that service */
+        nlohmann::json *item = &cubesConfig[cubeName];
+        bool deleted = false;
+        for (int i = 2; i < resItem.size() - 1; i++) {
+          auto *el = &(*item)[resItem[i]];
+          if (el->is_array()) { /* If it is an element of an array, treat it differently */
+            for (int j = 0; j < el->size(); j++) {/* Find the interested element of the array */
+              if ((*el)[j]["name"] == resItem[i + 1]) {
+                if (i == resItem.size() - 2) { /* If the whole element of the array is to be deleted */
+                  el->erase(j);
+                  deleted = true; /* Immediately delete it to avoid losing the reference */
+                  if (el->empty()) { /* If the array is empty, delete it */
+                    item->erase(resItem[i]);
+                  }
+                } else { /* Otherwise delete the element of the array element */
+                  el = &(*item)[resItem[i]][i];
+                  i++;
+                }
+                break;
+              }
+            }
+          }
+          item = el;
+        }
+        if (!deleted) {
+          item->erase(resItem[resItem.size() - 1]);
+        }
+      }
+      break;
+    }
+  }
+
+  if (!RestServer::startup) {
+    toSave++;
+    data_cond.notify_one();
+  }
+}
+
+void Resource::SaveToFile(const std::string& path) {
+  while (true) {
+    std::unique_lock<std::mutex> uniqueLock(mutex);
```

Comment:
Change name.
You don't have to call "lock" a variable that is a lock. It means nothing.
You have to tell people what is this lock doing: what is it protecting? E.g. dumpfile_lock looks more understandable. Adding a comment when you define the variable looks even better.